src/main/java/openfasttrack/importer/specobject/SpecObjectHandlerConfigBuilder.java

```diff
@@ -0,0 +1,169 @@
+package openfasttrack.importer.specobject;
+
+/*-
+ * #%L
+ * OpenFastTrack
+ * %%
+ * Copyright (C) 2016 - 2017 hamstercommunity
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+import java.util.logging.Logger;
+
+import openfasttrack.core.SpecificationItemId;
+import openfasttrack.core.SpecificationItemId.Builder;
+import openfasttrack.core.xml.CallbackContentHandler;
+import openfasttrack.core.xml.EventContentHandler;
+import openfasttrack.core.xml.event.Attribute;
+import openfasttrack.importer.ImportEventListener;
+import openfasttrack.importer.ImporterException;
+
+public class SpecObjectHandlerConfigBuilder
+{
+    private static final Logger LOG = Logger
+            .getLogger(SpecObjectHandlerConfigBuilder.class.getName());
+
+    private static final String DOCTYPE_ATTRIBUTE_NAME = "doctype";
+    private final CallbackContentHandler handler;
+    private String defaultDoctype;
+    private final String fileName;
+    private final ImportEventListener listener;
+
+    private boolean validRootElementFound = false;
+
+    private boolean needsCoverageSection = false;
+    private boolean providesCoverageSection = false;
+    private boolean dependenciesSection = false;
+    private Builder providesCoverageIdBuilder = null;
+    private Builder idBuilder = null;
+
+    public SpecObjectHandlerConfigBuilder(final String fileName, final ImportEventListener listener)
+    {
+        this.fileName = fileName;
+        this.listener = listener;
+
+        this.handler = new CallbackContentHandler();
+    }
+
+    public EventContentHandler build()
```

Comment:
This method is quite overloaded. I suggest to refactor.

Reply:
Done.